src/dataSource.js

```diff
@@ -96,7 +98,26 @@ class DataSource {
    * @returns {Array|Object}
    */
   getAtRow(row) {
-    return this.data[row];
+    let sourceDataRow = null;
+
+    // TODO: needs to be discussed
+
+    if (isObject(this.data[row])) {
+      sourceDataRow = {};
+
+      Object.keys(this.data[row]).map((key) => {
```

Comment:
I would suggest using `forEach` instead of `map` here. `map` will return an object and then the garbage collector will destroy it anyway.